File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

class Solution {
		public static List<Integer> solution(String[] name, int[] yearning, String[][] photo) {
			int[] answer = {};
			List<Integer> result = new ArrayList<>();

			Map<String, Integer> map = new HashMap<>();

			for (int i = 0; i < name.length; i++) {
				map.put(name[i], yearning[i]);
			}

			for (int i = 0; i < photo.length; i++) {
				int value = 0;
				for (String str : photo[i]) {
					Integer yearningValue = map.getOrDefault(str, 0);
					value += yearningValue.intValue();
				}
				result.add(value);
			}

			return result;
		}
	}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    let answer = [];

    let map = new Map();

    for (let i = 0; i < name.length; i++) {
        map.set(name[i], yearning[i]);
    }

    for (let i = 0; i < photo.length; i++) {
        let value = 0;
        for (const str of photo[i]) {

            let score = map.get(str);
            if (score !== undefined) {
                value += score;
            }
        }
        answer[i] = value;
    }

    return answer;
}